engine/table/src/main/java/io/deephaven/engine/table/impl/locations/impl/AbstractTableLocation.java

```diff
@@ -73,6 +73,14 @@ protected void destroy() {
         };
     }
 
+    /**
+     * This method is called before every public method so that the child classes can initialize the location lazily, if
+     * necessary.
```

Comment:
This doesn't seem true; for example, you are not calling it on `getColumnLocation` (on purpose I suspect).